src/cascadia/LocalTests_TerminalApp/TerminalApp.LocalTests.vcxproj

```diff
@@ -0,0 +1,150 @@
+
+  <PropertyGroup>
+  <!-- Manually change our outdir to be in a subdirectory. We don't really want
+    to put our output in the bin root, because if we do, we'll copy
+    TerminalApp.winmd to the bin root, and then every subsequent mdmerge step
+    (in _any_ cppwinrt project) will automatically try to pick up
+    TerminalApp.winmd as a dependency (which is just wrong). This MUST be done
+    after importing common.build.post.props-->
+    <OutDir>$(OpenConsoleDir)\bin\$(Platform)\$(Configuration)\$(ProjectName)\</OutDir>
+    <IntDir>$(OpenConsoleDir)\obj\$(Platform)\$(Configuration)\$(ProjectName)\</IntDir>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <_CppWinrtBinRoot>&quot;$(OpenConsoleDir)$(Platform)\$(Configuration)\&quot;</_CppWinrtBinRoot>
+    <!-- From Microsoft.UI.Xaml.targets -->
+    <Native-Platform Condition="'$(Platform)' == 'Win32'">x86</Native-Platform>
+    <Native-Platform Condition="'$(Platform)' != 'Win32'">$(Platform)</Native-Platform>
+    <_MUXBinRoot>&quot;$(OpenConsoleDir)packages\Microsoft.UI.Xaml.2.2.190611001-prerelease\runtimes\win10-$(Native-Platform)\native\&quot;</_MUXBinRoot>
+  </PropertyGroup>
+
+<!-- We actually can just straight up reference MUX here, it's fine -->
+  <Import Project="..\..\..\packages\Microsoft.UI.Xaml.2.2.190611001-prerelease\build\native\Microsoft.UI.Xaml.targets" Condition="Exists('..\..\..\packages\Microsoft.UI.Xaml.2.2.190611001-prerelease\build\native\Microsoft.UI.Xaml.targets')" />
+  <Import Project="..\..\..\packages\Microsoft.Toolkit.Win32.UI.XamlApplication.6.0.0-preview6.2\build\native\Microsoft.Toolkit.Win32.UI.XamlApplication.targets" Condition="Exists('..\..\..\packages\Microsoft.Toolkit.Win32.UI.XamlApplication.6.0.0-preview6.2\build\native\Microsoft.Toolkit.Win32.UI.XamlApplication.targets')" />
+
+  <!-- This project will generate individual sxs manifests for each of our winrt libraries -->
+  <Import Project="$(OpenConsoleDir)\build\rules\GenerateSxsManifestsFromWinmds.targets" />
+
+  <!-- This is important: actually add the _LocalTestsGenerateCombinedManifests
+    target to the list of targets to run. -->
+  <PropertyGroup>
+    <BeforeLinkTargets Condition="'$(WindowsTargetPlatformVersion)' &gt;= '10.0.18362.0'">
+      $(BeforeLinkTargets);
+      _LocalTestsGenerateCombinedManifests;
+      _LocalTestsBuildAppxManifest;
+    </BeforeLinkTargets>
+  </PropertyGroup>
+
+  <!-- Step 1: Combine all our SxS manifests into a single SxS manifest. TAEF
+  needs us to specify a single activation context at runtime, so we need a
+  single file with all our types in it.-->
+  <Target Name="_LocalTestsGenerateCombinedManifests"
+          Inputs="@(_ConsoleWinmdManifest)"
+          Outputs="$(OutDir)$(TargetName).manifest"
+          DependsOnTargets="_ConsoleGenerateAdditionalWinmdManifests">
+
+    <Exec Command="mt.exe -manifest @(_ConsoleWinmdManifest, ' -manifest ') -out:$(OutDir)$(TargetName).manifest" />

```

Comment:
i am so happy to see you writing custom msbuild targets
it is like an infection

Reply:
I'm not pleased that this is something we have to do at all, but I concur it's a lot smoother than xcopying files all over